File: 1497-check-if-array-pairs-are-divisible-by-k/1497-check-if-array-pairs-are-divisible-by-k.cpp
class Solution {
public:
    bool canArrange(vector<int>& arr, int k) {
        vector<int>hs(k,0);
        for(int &it:arr){
            int rem=(it%k+k)%k;
            hs[rem]++;
        }
        if(hs[0]%2)return false;
        for(int i=1;i<=k/2;i++){
            if(hs[k-i]!=hs[i])return false;
        }
        return true;
    }
};

File: 1497-check-if-array-pairs-are-divisible-by-k/1497-check-if-array-pairs-are-divisible-by-k.py
class Solution:
    def canArrange(self, arr, k):
        remainder_count = [0] * k
        for num in arr:
            rem = (num % k + k) % k  
            remainder_count[rem] += 1
        if remainder_count[0] % 2 != 0:
            return False  
        
        for i in range(1, k // 2 + 1):
            if remainder_count[i] != remainder_count[k - i]:
                return False 
        
        return True